sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/EncryptionCosmosAsyncDatabase.java

```diff
@@ -0,0 +1,392 @@
+                                                                                      String partitionKeyPath,
+                                                                                      ThroughputProperties throughputProperties) {
+        return this.cosmosAsyncDatabase.createContainerIfNotExists(id, partitionKeyPath, throughputProperties).map(cosmosContainerResponse -> {
+            EncryptionContainerResponse encryptionContainerResponse =
+                new EncryptionContainerResponse(cosmosContainerResponse,
+                    new EncryptionCosmosAsyncContainer(cosmosAsyncDatabase.getContainer(cosmosContainerResponse.getProperties().getId()), this.encryptionCosmosAsyncClient));
+            return encryptionContainerResponse;
+        });
+    }
+
+    /**
+     * Gets a CosmosAsyncClientEncryptionKey object without making a service call
+     *
+     * @param id id of the clientEncryptionKey
+     * @return Cosmos ClientEncryptionKey
+     */
+    public CosmosAsyncClientEncryptionKey getClientEncryptionKey(String id) {
+        return this.cosmosAsyncDatabase.getClientEncryptionKey(id);
+    }
+
+    /**
+     * Reads all cosmos client encryption keys in a database.
+     * <p>
+     * After subscription the operation will be performed. The {@link CosmosPagedFlux} will
+     * contain one or several feed response of the read cosmos client encryption keys. In case of
+     * failure the {@link CosmosPagedFlux} will error.
+     *
+     * @return a {@link CosmosPagedFlux} containing one or several feed response pages of the
+     * read cosmos client encryption keys or an error.
+     */
+    public CosmosPagedFlux<CosmosClientEncryptionKeyProperties> readAllClientEncryptionKeys() {
+        return this.cosmosAsyncDatabase.readAllClientEncryptionKeys();
+    }
+
+    /**
+     * Creates a client encryption key after subscription the operation will be performed. The
+     * {@link Mono} upon successful completion will contain a single resource
+     * response with the created client encryption key. In case of failure the {@link Mono} will
+     * error.
+     *
+     * @param clientEncryptionKeyId     Client Encryption Key id.
+     * @param encryptionAlgorithm       Encryption Algorithm.
+     * @param encryptionKeyWrapMetadata EncryptionKeyWrapMetadata.
+     * @return an {@link Mono} containing the single resource response with the
+     * created cosmos client encryption key or an error.
+     */
+    public Mono<CosmosClientEncryptionKeyResponse> createClientEncryptionKey(String clientEncryptionKeyId,
+                                                                             String encryptionAlgorithm,
+                                                                             EncryptionKeyWrapMetadata encryptionKeyWrapMetadata) {
+        if (StringUtils.isEmpty(clientEncryptionKeyId)) {
+            return Mono.error(new IllegalArgumentException("clientEncryptionKeyId is null or empty"));
```

Comment:
any reason we are returning Mono of error instead of throwing `IllegalArgumentException` directly?